From a480a9659fb0cab1c44006f6c06013e6e7f78948 Mon Sep 17 00:00:00 2001
From: Martin Sehnoutka <msehnout@redhat.com>
Date: Fri, 8 Apr 2016 15:05:53 +0200
Subject: [PATCH 7/7] vsftpd-2.2.2-syslog

---
 logging.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/logging.c b/logging.c
index 99671b4..c4461f7 100644
--- a/logging.c
+++ b/logging.c
@@ -32,7 +32,7 @@ vsf_log_init(struct vsf_session* p_sess)
 {
   if (tunable_syslog_enable || tunable_tcp_wrappers)
   {
-    vsf_sysutil_openlog(1);
+    vsf_sysutil_openlog(0);
   }
   if (!tunable_xferlog_enable && !tunable_dual_log_enable)
   {
-- 
2.5.5

