--- vsftpd-2.0.5/defs.h.uname_size	2007-04-13 15:15:54.000000000 +1000
+++ vsftpd-2.0.5/defs.h	2007-04-13 15:19:14.000000000 +1000
@@ -6,7 +6,7 @@
 #define VSFTP_COMMAND_FD        0
 
 #define VSFTP_PASSWORD_MAX      128
-#define VSFTP_USERNAME_MAX      32
+#define VSFTP_USERNAME_MAX      128
 #define VSFTP_MAX_COMMAND_LINE  4096
 #define VSFTP_DATA_BUFSIZE      65536
 #define VSFTP_DIR_BUFSIZE       16384
