--- vsftpd-1.0.1/vsftpd.conf.rh	Mon Jul 30 17:51:07 2001
+++ vsftpd-1.0.1/vsftpd.conf	Wed Nov 28 14:38:36 2001
@@ -7,14 +7,14 @@
 anonymous_enable=YES
 #
 # Uncomment this to allow local users to log in.
-#local_enable=YES
+local_enable=YES
 #
 # Uncomment this to enable any form of FTP write command.
-#write_enable=YES
+write_enable=YES
 #
 # Default umask for local users is 077. You may wish to change this to 022,
 # if your users expect that (022 is used by most other ftpd's)
-#local_umask=022
+local_umask=022
 #
 # Uncomment this to allow the anonymous FTP user to upload files. This only
 # has an effect if the above global write enable is activated. Also, you will
@@ -46,7 +46,7 @@
 #xferlog_file=/var/log/vsftpd.log
 #
 # If you want, you can have your log file in standard ftpd xferlog format
-#xferlog_std_format=YES
+xferlog_std_format=YES
 #
 # You may change the default value for timing out an idle session.
 #idle_session_timeout=600
@@ -98,3 +98,8 @@
 # the presence of the "-R" option, so there is a strong case for enabling it.
 #ls_recurse_enable=YES
 
+pam_service_name=vsftpd
+userlist_enable=YES
+#enable for standalone mode
+listen=YES
+tcp_wrappers=YES
--- vsftpd-1.2.1/RedHat/vsftpd.log.rh	2004-05-19 17:17:21.000000000 -0400
+++ vsftpd-1.2.1/RedHat/vsftpd.log	2004-05-19 17:17:27.000000000 -0400
@@ -1,4 +1,4 @@
-/var/log/vsftpd.log {
+/var/log/xferlog {
     # ftpd doesn't handle SIGHUP properly
     nocompress
     missingok
