--- vsftpd-2.0.1/postlogin.c.suniq	2005-08-03 22:18:03.719180808 +0530
+++ vsftpd-2.0.1/postlogin.c	2005-08-03 22:17:25.910928536 +0530
@@ -1013,7 +1013,6 @@
     struct mystr resp_str = INIT_MYSTR;
     str_alloc_text(&resp_str, "FILE: ");
     str_append_str(&resp_str, p_filename);
-    vsf_cmdio_write_str(p_sess, FTP_DATACONN, &resp_str);
     remote_fd = get_remote_transfer_fd(p_sess, str_getbuf(&resp_str));
     str_free(&resp_str);
   }
