diff -pruN vsftpd-2.1.0.orig/prelogin.c vsftpd-2.1.0/prelogin.c
--- vsftpd-2.1.0.orig/prelogin.c	2009-04-12 16:00:09.000000000 +0200
+++ vsftpd-2.1.0/prelogin.c	2009-04-12 16:13:57.000000000 +0200
@@ -217,6 +217,7 @@ handle_user_command(struct vsf_session* 
     if ((located && tunable_userlist_deny) ||
         (!located && !tunable_userlist_deny))
     {
+      check_login_delay();
       vsf_cmdio_write(p_sess, FTP_LOGINERR, "Permission denied.");
       check_login_fails(p_sess);
       if (tunable_userlist_log)
