From cdcb1c5f660fb4f72b4896f5145a34e9dd158252 Mon Sep 17 00:00:00 2001
From: Martin Sehnoutka <msehnout@redhat.com>
Date: Thu, 17 Mar 2016 12:00:34 +0100
Subject: [PATCH 26/26] Applied vsftpd-2.0.5-fix_qm.patch

---
 ls.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ls.c b/ls.c
index f18791d..de68c30 100644
--- a/ls.c
+++ b/ls.c
@@ -459,7 +459,8 @@ vsf_filename_passes_filter(const struct mystr* p_filename_str,
         must_match_at_current_pos = 0;
       }
     } while (locate_result.found &&
-             str_getlen(&name_remain_str) > 0 && last_token != '*');
+             str_getlen(&name_remain_str) > 0 &&
+             last_token != '*' && last_token != '?');
   }
   /* Any incoming string left means no match unless we ended on the correct
    * type of wildcard.
-- 
2.5.0

